core/src/main/java/com/linecorp/armeria/common/ResponseEntity.java

```diff
@@ -62,6 +53,38 @@ static <T> ResponseEntity<T> of(ResponseHeaders headers, @Nullable T content, Ht
         return new DefaultResponseEntity<>(headers, content, trailers);
     }
 
+    /**
+     * Returns a newly created {@link ResponseEntity} with the specified {@link HttpStatus}.
+     */
+    static <T> ResponseEntity<T> of(HttpStatus status) {
+        return of(ResponseHeaders.of(status));
+    }
+
+    /**
+     * Returns a newly created {@link ResponseEntity} with the specified {@link HttpStatus} and
+     * {@code content}.
+     */
+    static <T> ResponseEntity<T> of(HttpStatus status, T content) {
+        return of(ResponseHeaders.of(status), content);
+    }
+
+    /**
+     * Returns a newly created {@link ResponseEntity} with the specified {@link HttpStatus},
+     * {@code content} and {@linkplain HttpHeaders trailers}.
+     */
+    static <T> ResponseEntity<T> of(HttpStatus status, @Nullable T content, HttpHeaders trailers) {
+        return of(ResponseHeaders.of(status), content, trailers);
+    }
+
+    /**
+     * Returns a newly created {@link ResponseEntity} with the specified {@code content} and
+     * {@link HttpStatus#OK} status.
+     */
+    static <T> ResponseEntity<T> of(T content) {
+        requireNonNull(content, "content");
+        return of(HttpStatus.OK, content);
```

Comment:
Is it good practice to reuse other `of()` methods, or should I just call the one that has the fewest layers on top of `DefaultResponseEntity`?

Reply:
> Is it good practice to reuse other of() methods,

I prefer calling other `of()` methods but I believe it depends on the situation. It's especially useful when we can dedup the logic by delegating the method call.

Additionally, I prefer having methods that have non-nullable parameters if possible. 
What do you think of changing the signature a bit?
```
    static <T> ResponseEntity<T> of(ResponseHeaders headers) {
        requireNonNull(headers, "headers");
        return new DefaultResponseEntity<>(headers, null, HttpHeaders.of());
    }

    static <T> ResponseEntity<T> of(ResponseHeaders headers, T content) {
        return of(headers, content, HttpHeaders.of());
    }

    static <T> ResponseEntity<T> of(ResponseHeaders headers, T content, HttpHeaders trailers) {
        requireNonNull(headers, "headers");
        requireNonNull(content, "content");
        requireNonNull(trailers, "trailers");
        return new DefaultResponseEntity<>(headers, content, trailers);
    }
```

Reply:
Sure, sounds good!